[HUDI-225] Create Hudi Timeline Server Fat Jar

leesf opened the pull request:
see jira:[HUDI-225](https://jira.apache.org/jira/projects/HUDI/issues/HUDI-225)

vinothchandar commented:
@bvaradar Tip: I have been using "squash and merge"  option which lets you control commit message and also squashes automatically

leesf commented:
@vinothchandar @bvaradar Thanks for the review, addressed the comments and squashed the commits. And two things to point out. First when running mvn notice:generate  from top-level module, the notice file(NOTICE.txt and HUDI_NOTICE.txt) in other modules(such as hudi-hadoop-mr and packaging/hudi-hadoop-mr-bundle) have been modified, so should we open anthor PR to update NOTICE file? And i am willing to do this work. Second. In windows environment, running mvn notice:generate gets the following exception:

> Failed to execute goal org.jasig.maven:maven-notice-plugin:1.1.0:generate (default-cli) on project hudi: Resource not found in file system, classpath or URL: file://F:/community
/incubator-hudi/release/config/license-mappings.xml

And it can be fix by removing _file://_ in configuration, and it works well in windows and linux. So should i file another jira ticket to track it?

bvaradar commented:
@leesf: As far as this PR goes, it looks good. We found mvn notice:generate to be noisy in that it perceives changes when there shouldn't be (depending in the maven setup). I had to update notices as a preparatory step for release. So, I wouldn't worry about the notice:generate changes on other modules. The other change w.r.t file:// would be good to have. Please take a look at POM for other modules to see if there are other such examples

leesf commented:
@bvaradar Thanks for your reminder, updated the PR. And will file another jira ticket to track mvn notice:generate issue in windows.
